asterius/src/Asterius/Builtins/Primitive.hs

```diff
@@ -0,0 +1,181 @@
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+-- | Webassembly implementations for primitive operations (@memcpy@, @memmove@,
+-- @memcmp@, and @memset@ variants). Each binding corresponds to a definition
+-- in the cbits of the @primitive@ package:
+--
+-- https://github.com/haskell/primitive/blob/c07823669e542399b7af11ffbf924d7106e3f145/cbits/primitive-memops.h#L1-L23
+module Asterius.Builtins.Primitive
+  ( primitiveImports,
+    primitiveCBits,
+  )
+where
+
+import Asterius.EDSL
+import Asterius.Types
+import qualified Data.ByteString as BS
+
+-- GEORGE: Maybe use this function in all Asterius.Builtin.* modules to save us
+-- from duplication and copy-paste errors?
+
+-- | Create a function import with internal name of the form
+-- @__asterius_<EXTERNAL-MODULE-NAME>_<EXTERNAL-BASE-NAME>@.
+mkImport ::
+  -- | External module name
+  BS.ByteString ->
+  -- | External base name
+  BS.ByteString ->
+  -- | Function type.
+  FunctionType ->
+  FunctionImport
+mkImport ext_mod_name ext_base_name fn_type =
+  FunctionImport
+    { internalName = "__asterius_" <> ext_mod_name <> "_" <> ext_base_name,
+      externalModuleName = ext_mod_name,
+      externalBaseName = ext_base_name,
+      functionType = fn_type
+    }
+{-# INLINE mkImport #-}
+
+-- -------------------------------------------------------------------------
+
+-- GEORGE: Note that Asterius uses 64 bits everywhere, but WebAssembly at the
+-- moment does not support I64. Yet, it DOES support F64. Hence, we convert
```

Comment:
This needs clarification. [WebAssembly *does* support `I64`](https://webassembly.github.io/spec/core/syntax/types.html), and `F64` is subject to floating-point normalisation, so certain values might be problematic.

Reply:
@ggreif You are right that certain values are problematic because of the representation (see https://github.com/tweag/asterius/blob/df1f5b1ba57619cc14369554eef2a323c0e9cef4/asterius/rts/rts.memory.mjs#L294-L313). We have no clear solution to the problem yet though, that's why I added these comment, lest we forget.

Reply:
I think I get what you mean, I keep forgetting your boundary condition, that you _always_ interface with some JS module. I live in a somewhat different world, where I can basically use all Wasm-permitted types when calling imports, also `I64`.

Reply:
> I live in a somewhat different world, where I can basically use all Wasm-permitted types when calling imports, also `I64`.

Indeed, I wish I could live in that world too! For now though you are right that this needed more attention. I opened a separate issue documenting this behavior (https://github.com/tweag/asterius/issues/682) instead of keeping this only in comments. Hopefully that clarifies things for others too :slightly_smiling_face: